mos-platform/common/c/stdio-full.c

```diff
@@ -367,110 +335,20 @@ FILE *freopen(const char *restrict filename, const char *restrict mode,
     return NULL;
   }
 
-  /* Flush buffer */
-  if (stream->status & FWRITE)
-    flush_buffer(stream);
-
-  if (filename == NULL) {
-    /* Attempt to change mode without closing stream */
-    switch (change_mode(stream, fmode)) {
-    case INT_MIN:
-      /* fail completely */
-      return NULL;
-
-    case 0:
-      /* fail; try close / reopen */
```

Comment:
This PR should have `[Breaking]` in its description, since this is a change to existing implementation-defined behavior.

Reply:
Since we're rebasing; this should be done on the individual commit that introduces the breaking change.

Reply:
Please read the new commit messages. If we split the first commit, the motivation could be become unclear.